diesel_migrations/migrations_internals/src/lib.rs

```diff
@@ -320,14 +321,33 @@ fn run_migration<Conn>(
 where
     Conn: MigrationConnection,
 {
-    conn.transaction(|| {
+    let mut run_migration = || {
         if migration.version() != "00000000000000" {
             try!(writeln!(output, "Running migration {}", name(&migration)));
         }
         try!(migration.run(conn));
         try!(conn.insert_new_migration(migration.version()));
         Ok(())
-    })
+    };
+
+    let run_in_transaction = migration
+        .metadata()
+        .and_then(|m| m.get("run_in_transaction"));
```

Comment:
The metadata key that diesel understands should probably be exported as constants somewhere.

Reply:
I meant publicly exported.

Reply:
Why?

Reply:
So that if someone else wants to look at the same metadata they don't have to duplicate the strings, they can import the constant from diesel, and for someone reading the code it's obvious how everything fits together. The constant would also serve as a convenient place to put the documentation.

Also, for someone working on diesel, it's more obvious that the constant is part of the public API (ie. it can't just be changed without breaking stuff) if it is actually part of the public API.